drivers/usb/device/usb_dc_stm32.c

```diff
@@ -323,6 +323,22 @@ static int usb_dc_stm32_init(void)
 	return 0;
 }
 
+
+/*
+ * The STM32L432 SoC needs to control the isolation of the USB features
+ * from VDDUSB. This is done through the PWR_CR2 bit USV. The STM32L4 HAL
+ * in stm32l4xx_ll_pwr.h checks for for the PWR_CR2_PVME1 bit, which is
+ * only available on Cat. 3 devices. It should directly check for
+ * PWR_CR2_USV instead, like done in stm32l4xx_hal_pwr_ex.c.
+ * This is likely a bug in the HAL, add a workaround until it is fixed.
```

Comment:
Thanks for reporting this, I've filed a bug in HAL (internal ref 47844).
Do you mind updating directly the LL file? This way the patch would also be work for other versions of L432 and for L442 SoC as well. Please don't forget to update Patch list section in ext/hal/st/stm32cube/stm32l4xx/README.

Reply:
Thanks @erwango for reporting the bug in HAL. I was not aware it is possible to patch HAL directly, it indeed looks a much better solution. I have done that, simplified the changes in drivers/usb/device/usb_dc_stm32.c, and just pushed the changes.